constrained-voucher.mkd

```diff
@@ -520,6 +506,92 @@ One key difference is that the names of the leaves in the YANG does not have a m
 
 Any POST request to the Registrar with resource /est/vs or /est/es returns a 2.05 response with empty payload. The client should be aware that the server may use a piggybacked CoAP response (ACK, 2.05) but may also respond with a separate CoAP response, i.e. first an (ACK, 0.0) that is an acknowledgement of the request reception followed by a (CON, 2.05) response in a separate CoAP message.
 
+# Raw Public Key Use Considerations {#rpk-considerations}
+
+There is a strong desire to be able to build constrained (IoT) devices which have a minimal amount of code, and which do a minimal amount of network traffic.
+Note that these are two goals, and these two goals may be at odds.
+
+One piece of code that is complex, hard to test, and never seems critical for application design is the PKIX certification validation code.
+This section explains how a device can be onboarded with the minimal amount of PKIX related code.
+
+## The IDevID artifact
+
+The BRSKI protocol is based upon the Pledge identifying itself with an RFC5280 format (X509v3, PKIX) certificate.
+The details of how it becomes an IDevID is defined in {{ieee802-1AR}}.
+While the Pledge device needs to have an IDevID object, it does not need to manipulate it.
+It can regard the IDevID as an opaque object which is placed into a DTLS setup.
+
+The IDevID certificate will have an associated private key.
+The Pledge will need to use that private key for the DTLS connection, and for signing of the voucher-request.
+However, none of these operations require any kind of manipulation of certificates.
+The private key can be loaded in whatever format is appropriate and efficient for the hardware.
+(EDNOTE: are there lwig references that would make sense here?)
+
+## The Registrar Trust Anchor
+
+When the Pledge first connects to the Registrar, the connection to the Registrar is provisional, as explained in section 5.6.2 of {{I-D.ietf-anima-bootstrapping-keyinfra}}.
+The Registrar provides its public key in a TLSServerCertificate, and the Pledge uses that to validate that integrity of the (D)TLS connection, but it does not validate the identity of
+the provided certificate.
+
+If instead of using a (TLSServer)Certificate of type X509 (see section 4.4.2 of {{RFC8446}}),
```

Comment:
This I don't understand. If the Pledge connects with an IDevID certificate, why would the Registrar respond with an RPK as its server identity?   I thought the idea was that the Pledge can "scan" the Registrar server cert for the public-key part and use that for asserting proximity and, later on, validation against the pinned RPK in the Voucher.

I assume here we don't want to introduce another RPK method that impacts the Registrar in this way?

Reply:
Ack: how would the registrar know what it should respond with, this early in the protocol?

Reply:
Update: if I read RFC 7250, it seems clear that a client (PLedge) already signals in its first DTLS Client Hello handshake message that it supports RPK for the handshake and the server signals its support back in Server Hello. That seems to solve the issue.

So it would be good to keep the different cases in mind when writing the text:
1. Regular DTLS handshake based on IDevID/Registrar certs
2. Regular DTLS handshake based on IDevID/Registrar certs - but the Pledge cannot do PKIX validation and can only "extract" the RPK (SubjectPublicKeyInfo) structure from the Registrar's cert chain, discarding the rest.
3. RPK DTLS handshake per RFC 7250

For case 1/2 the benefit is that Registrar can get the MASA URI from the IDevID. For case 3 this isn't possible, so the Registrar must have a-priori a list of the Pledges to enroll and their public key identities along with info on which vendor/MASA each Pledge uses.